src/modeler/AutoRest.Swagger/Validation/DescriptionRequired.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class DescriptionRequired : TypedRule<SwaggerObject>
+    {
+        public override bool IsValid(SwaggerObject entity)
+        {
+            bool valid = false;
+
+            if (entity != null)
+            {
+                valid = entity.Description != null || !string.IsNullOrEmpty(entity.Reference);
+            }
+
+            return valid;
+        }
```

Comment:
Just curious about the consistency...

Some `Rules` you create a `bool valid` and set the value based on the check, and return.
``` csharp
public override bool IsValid(SwaggerObject entity)
{
    bool valid = false;

    if (entity != null)
    {
        valid = entity.Description != null || !string.IsNullOrEmpty(entity.Reference);
    }

    return valid;
}
```

and in some you just return the boolean state of the check.

``` csharp
public override bool IsValid(SwaggerObject entity)
{  
  return entity.Description != null || !string.IsNullOrEmpty(entity.Reference);
}
```

Personally, I'm happier seeing the terse version. 

and for even greater simplicity, the whole class can get distilled to :

``` csharp
public class DescriptionRequired : TypedRule<SwaggerObject>
{
  public override bool IsValid(SwaggerObject entity) => entity.Description != null || !string.IsNullOrEmpty(entity.Reference);

  public override ValidationExceptionName Exception => ValidationExceptionName.DescriptionRequired;
}
```